File: IssueManager.Client.Vue3/src/service/TicketService.js
const API_ENDPOINT = "https://605a5f9327f0050017c04d9c.mockapi.io/api/v1"
export default {
    getTickets: async function () {
        return await fetch(`${API_ENDPOINT}/ticket`)
    },
    getTicket: async (id) => {
        return await fetch(`${API_ENDPOINT}/ticket/${id}`)
    },
    searchTicket: async (text) => {
        return await fetch(`${API_ENDPOINT}/ticket?search=${text}`)
    },
    updateTicket: async (ticket, id) => await fetch(`${API_ENDPOINT}/ticket/${id}`, {
        method: 'PUT',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(ticket),
    }),
    createTicket: async (ticket) => await fetch(`${API_ENDPOINT}/ticket`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(ticket),
    })
}

File: IssueManager.ThirdParty.Api/Interface/TicketService.cs
using System;
using System.Net.Http;
using System.Collections.Generic;
using IssueManager.Domain.Model;
using System.Threading.Tasks;
using System.Net.Http.Json;

namespace IssueManager.ThirdParty.Api.Interface
{
    public class TicketService : ITicketService
    {
        static readonly string API_ENDPOINT = "https://605a5f9327f0050017c04d9c.mockapi.io/api/v1";

        readonly HttpClient client = new HttpClient { BaseAddress = new Uri(API_ENDPOINT) };

        public async Task<List<Ticket>> GetTickets()
        {
            return await client.GetFromJsonAsync<List<Ticket>>("/ticket");
        }

        public async Task<Ticket> GetTicket(int id)
        {
            return await client.GetFromJsonAsync<Ticket>($"/ticket/{id}");
        }

        public async Task<List<Ticket>> SearchTickets(string searchText)
        {
            return await client.GetFromJsonAsync<List<Ticket>>($"/ticket?search={searchText}");
        }

        public async Task<Ticket> UpdateTicket(Ticket ticket)
        {
            var response = await client.PutAsJsonAsync<Ticket>($"/ticket/{ticket.Id}", ticket);
            return await response.Content.ReadFromJsonAsync<Ticket>();
        }
        public async Task<Ticket> CreateTicket(Ticket ticket)
        {
            var response = await client.PostAsJsonAsync<Ticket>($"/ticket", ticket);
            return await response.Content.ReadFromJsonAsync<Ticket>();
        }

        public async Task<Ticket> RestoreTicket(Ticket ticket)
        {
            ticket.IsDeleted = false;
            var response = await client.PutAsJsonAsync<Ticket>($"/ticket/{ticket.Id}", ticket);
            return await response.Content.ReadFromJsonAsync<Ticket>();
        }
    }
}
